api/lib/domain/services/authentication/oidc-authentication-service.js

```diff
@@ -50,14 +52,34 @@ class OidcAuthenticationService {
     this.authenticationUrl = authenticationUrl;
     this.authenticationUrlParameters = authenticationUrlParameters;
     this.userInfoUrl = userInfoUrl;
+
+    const missingRequiredProperties = [];
+    this.requiredProperties.forEach((requiredProperty) => {
+      if (lodash.isNil(config[this.configKey][requiredProperty])) {
+        missingRequiredProperties.push(requiredProperty);
+      }
+    });
+
+    const isEnabledInConfig = config[this.configKey]?.isEnabled;
```

Comment:
Question : on a vraiment des cas ou this.configKey serait non renseigné ? Parce que si on fait par exemple un new de  fwb-oidc-authentication-service.js, on voit que les valeurs du constructeur son renseignée en dur.

est-ce qu'on devrait pas être plus intransigeant ? et si config[this.configKey] stopper et faire un logger comme plus bas ?


deuxième point : c'est pas la première fois que je me dis que fwb-oidc-authentication-service.js mélange un peu service et objet du domaine.

quand on fait un new, on a plein de valeurs en dur dans le constructeur mais comme c'est un service on a aussi des méthodes comme saveIdToken qui enregistre en base de données.

j'ai pas de réponse de code mais je trouve qu'il y a un bad smell là dedans

troisième point : 
On a aussi une liste de  requiredProperties: ['clientId', 'clientSecret', 'authenticationUrl', 'userInfoUrl', 'tokenUrl', 'logoutUrl'], qui sont en dur et qui sont importantes (ainsi que organizationName ou autre)

On a bien le test d'acceptance sur le get identity provider  mais est-ce qu'un test unitaire de ce service en vérifiant par que requiredProperties contient bien les données ?

du genre on fait un const toto = new fwb-oidc-authentication-service.js
et puis on vérifie que 
toto.requiredProperties == ['clientId', 'clientSecret', 'authenticationUrl', 'userInfoUrl', 'tokenUrl', 'logoutUrl']
toto.organizationName == 'fwb bla bla'

sorry, c'est du pseudo code un peu moche ^^

Reply:
@annemarie35 pour le premier point je suis complètement d'accord d'être plus intransigeant et je vais faire ce que tu indiques. J'avais hésité entre `config[this.configKey].isEnabled` et `config[this.configKey].isEnabled` et finalement opté pour `config[this.configKey]?.isEnabled` car cela permettait de ne pas modifier de nombreux tests pour y ajouter de la config avec une clé `configKey` dedans.

Reply:
Pour le 3ème point : d'accord qu'il serait bon d'avoir des tests unitaires des `*bOidcAuthenticationService.js`. Je vais en ajouter.

Reply:
La présence de la propriété `configKey` est maintenant testée dans le constructeur de `OidcAuthenticationService`, avant `isEnabledInConfig` et `isConfigValid`.